lms/djangoapps/courseware/courses.py

```diff
@@ -99,7 +109,7 @@ def get_course_by_id(course_key, depth=0):
         raise Http404(u"Course not found: {}.".format(six.text_type(course_key)))
 
 
-def get_course_with_access(user, action, course_key, depth=0, check_if_enrolled=False, check_survey_complete=True):
+def get_course_with_access(user, action, course_key, depth=0, check_if_enrolled=False, check_survey_complete=True, check_if_authenticated=False):
```

Comment:
You'll see a new check_if_authenticated argument in these functions.  This is because we want the micro-frontend to explicitly check to see if a user is authenticated.  Prior to this, this was done implicitly by checking if the user was enrolled.  If a user is anonymous, they therefore can't be enrolled.  In the micro-frontend, though, we will ultimately want to be able to tell these two situations apart because we need to display different messaging for them.  It defaults to false, meaning that existing behavior/calls to `get_course_with_access` won't be affected.